graphql/util.go

```diff
@@ -0,0 +1,18 @@
+package graphql
+
+import (
+	"fmt"
+	"time"
+)
+
+// GetFormattedDuration returns a time.Duration type in "33h 33m 33s" format
+func GetFormattedDuration(d time.Duration) *string {
```

Comment:
Why not use the String() function for durations?

Reply:
i googled formatting a Duration type and this was the 4 month old stackoverflow answer

Reply:
That seems like the wrong answer since go has a built-in format function that basically does this

Reply:
I would also use .String(). 

Functions like this would need a lot of tests, also I would avoid exporting it if possible. 

Looks like this doesn't handle durations <1h particularly well.

Reply:
go does have great built in support for Duration type, however does not do exactly what is needed. 

`time.ParseDuration` returns the format `1h1m45s` which is almost what we need but without the spaces. 

formatting the parsed duration using Duration methods with something like `fmt.Sprintf("%.0fh %.0fm %.0fs", parsedDuration.Hours(), parsedDuration.Minutes(), parsedDuration.Seconds())` returns correct format but the entire time in hours, the entire time in minutes, and entire time in seconds. 

therefore achieving the desired format requires this determine hours, subtract and then determine minutes and so on approach described in the stackoverflow answer. 

please correct me if i'm wrong but i think the SO answer is right.

Reply:
I think all of these solutions handle cases less than 1 hour poorly. See

https://play.golang.org/p/jWFy7wQOu4O

Why are the spaces required?

Reply:
would not be difficult to remove hours or minutes from format if they are < 1

spaces are required because of design but i would be fine with doing without the spaces

Reply:
lets try and do it without spaces :)